### DISCUSSION TITLE
Long boot time with opensnitch service enabled

### DISCUSSION THREAD

**Discussion by akronym0** [evilsocket/opensnitch#953] [Q&A]
Normally I hiberrnate/resume but recently I rebooted a lot for issue #946 and I noticed something odd.

When the OpenSnitch service is enabled, it takes roughly two minutes to boot into the GUI (during that time only a black screen is visible).

After

`sudo systemctl disable opensnitch.service`

booting into the GUI is near instant (that is, the black screen is only visibile for 1 or 2 seconds).

I have checked the boot log with journalctl and also the OpenSnitch log, but I could not see anything that would explain the long delay. I also disabled many other services, but only disabling OpenSnitch fixes the delay.

I have also check systemd-analyze blame and critical-chain. But nothing there exlains the two minute "blackout".

Does anybody have an explanation?

Linux Mint 21.1, Kernel 5.19, OpenSnitch 1.6rc5

A service file like this also fixes the boot issue:

```
[Unit]
Description=Application firewall OpenSnitch
Documentation=https://github.com/gustavo-iniguez-goya/opensnitch/wiki
Wants=multi-user.target
After=multi-user.target

[Service]
Type=simple
PermissionsStartOnly=true
ExecStartPre=/bin/mkdir -p /etc/opensnitchd/rules
ExecStart=/usr/bin/opensnitchd -rules-path /etc/opensnitchd/rules
Restart=always
RestartSec=30
TimeoutStopSec=10

[Install]
WantedBy=graphical.target
```

 (But it is probably not correct?)

**Comment by gustavo-iniguez-goya** [comment]
This is commonly caused when some binaries are blocked: https://github.com/evilsocket/opensnitch/wiki/Getting-started#notes-
Specially xbrlapi, dirmngr

A rule to allow connections `To this IP: 127.0.0.1` should fix this problem.
If that doesn't work, and the `DefaultAction` value in /etc/opensnitchd/default-config.json is set to `deny`,  could you set it to `allow` and see if the delay is fixed?

> A service file like this also fixes the boot issue:

If I'm not wrong, the only problem is that it starts once the graphical target is up and running, so you would miss previous connections.
